Formula/idris2.rb

```diff
@@ -1,31 +1,98 @@
 class Idris2 < Formula
   desc "Pure functional programming language with dependent types"
   homepage "https://www.idris-lang.org/"
-  url "https://github.com/idris-lang/Idris2/archive/v0.5.1.tar.gz"
-  sha256 "da44154f6eba5e22ec5ac64c6ba2c28d2df0a57cf620c5b00c11adb51dbda399"
   license "BSD-3-Clause"
   revision 2
   head "https://github.com/idris-lang/Idris2.git", branch: "main"
 
+  stable do
+    url "https://github.com/idris-lang/Idris2/archive/v0.5.1.tar.gz"
+    sha256 "da44154f6eba5e22ec5ac64c6ba2c28d2df0a57cf620c5b00c11adb51dbda399"
+
+    # Fix loading of dylibs on Monterey. Remove in the next release.
+    # Backport of https://github.com/idris-lang/Idris2/pull/2078
+    on_monterey :or_newer do
+      patch do
+        url "https://github.com/idris-lang/Idris2/commit/36403cbc3dab4ecadd8d4f8c9280ef890de09700.patch?full_index=1"
+        sha256 "c80dd2e96f03eab8446fdcb84d836e3792eb0eb4dbf5e644aec3b8ab4a1eb2f6"
+      end
+      patch :DATA
+    end
+
+    # Fix detection of Apple Silicon. Remove in the next release
+    patch do
+      on_arm do
+        url "https://github.com/idris-lang/Idris2/commit/21ca9066f13e3b9f8c6be28774a91f43bb68d51d.patch?full_index=1"
+        sha256 "468b0afc6f81b7709ba1cfd63d49e9183b5b3ba718e0af774405d3b750f2616d"
+      end
+    end
+  end
+
   bottle do
     sha256 cellar: :any,                 big_sur:      "39fe90502cd4a8a064b4f69bcc5b4e99295826cad363092b47d3353ed41db2da"
     sha256 cellar: :any,                 catalina:     "f29bcc24fc1a0581eb3f17823641c5444acc4df10e93c17eae573deab9a18fa7"
     sha256 cellar: :any_skip_relocation, x86_64_linux: "fdd19a501c9b80da87fd4519ed43e9ed31647e17e03d582392d2b3d1517f7c74"
   end
 
-  depends_on "coreutils" => :build
   depends_on "gmp" => :build
-  depends_on "chezscheme"
-  uses_from_macos "zsh" => :build, since: :mojave
+
+  on_high_sierra :or_older do
+    depends_on "zsh" => :build
+  end
```

Comment:
I think build may work with `bash` on Linux and `zsh` was added to avoid `bash + coreutils` on macOS, but will need to check results.